<launch>
  <!-- 以gazebo坐标系为参考系 -->
  <!-- <arg name="x_pos" default="-9.0"/> -->
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <!-- <arg name="y_pos" default="8.0"/> -->
  <arg name="z_pos" default="0.001"/>
  <!-- <arg name="yaw_angle" default="-1.5708"/> -->
  <arg name="yaw_angle" default="0.0"/>
  <!-- 启动仿真界面,如果不对仿真环境作修改，可以不显示gazebo -->
  <arg name="gui" default="false"/>

  <!-- load model -->
  <!-- 将 Urdf 文件的内容加载到参数服务器 -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find chibot_simu)/urdf/chibot_model.urdf.xacro'"/>
  <!-- 发布融合雷达坐标系 -->
  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_laser" args="0.0 0.00 0.1347 0 0 0  base_link laser" /> laser -->
  <node pkg="tf" type="static_transform_publisher" name="base_to_laser" args="0.0 0.00 0.1347 0 0 0  /base_link /laser 50.0" />
  <!-- 发布超声波坐标系 -->
  <!-- 左前 -->
  <!-- x偏移量 y偏移量 z偏移量 z偏航角度 y俯仰角度 x翻滚角度 父级坐标系 子级坐标系 -->
  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="ultra_left_front" args="0.3444 0.2678 0.072 0.7854 0 0  base_link ultra_left_front_link" /> ultra_left_front -->
  <!-- <node name="static_tf" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 /map /base_link 100" /> -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="ultra_left_front" args="0.3444 0.2678 0.072 0.7854 0 0  /base_link /ultra_left_front_link 100" /> -->
  <!-- 右前 -->
  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="ultra_right_front" args="0.3444 -0.2678 0.072 -0.7854 0 0  base_link ultra_right_front_link" /> ultra_right_front -->
  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="ultra_right_front" args="0.3444 -0.2678 0.072 -0.7854 0 0  base_link ultra_right_front_link" /> ultra_right_front -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="ultra_right_front" args="0.3444 -0.2678 0.072 -0.7854 0 0  /base_link /ultra_right_front_link 100" /> -->

  <!-- 订阅话题：/joint_states，发布小车TF信息-->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
   <param name="publish_frequency" type="double" value="50.0" />
  </node>
  <!-- 读取urdf信息(robot_description) 发布话题：/joint_states-->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
   <param name="use_gui" value="false" />
  </node>

  <!-- load world -->
  <!-- 启动 gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find chibot_simu)/worlds/complex.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- show model in gazebo -->
<!-- 
    在 Gazebo 中加载一个机器人模型，该功能由 gazebo_ros 下的 spawn_model 提供:
    -urdf 加载的是 urdf 文件
    -model chibot 模型名称是 chibot
    -param robot_description 从参数 robot_description 中载入模型
    -x 模型载入的 x 坐标
    -y 模型载入的 y 坐标
    -z 模型载入的 z 坐标
-->
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_model"
        args="-urdf -model chibot -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -Y $(arg yaw_angle) -param robot_description"/>

  <!-- 数据融合 -->
  <!-- <node pkg="chibot_simu" type="chibot_bridge_node" name="chibot_bridge" /> -->
</launch>
